src/device/onLoad.js

```diff
@@ -248,41 +248,48 @@ realityEditor.device.onload = async function () {
+            anyChanged = true;
+            toggleCloudUrl.onTextCallback(`https://${settings.serverUrl}/stable` +
+                `/n/${settings.networkUUID}` +
+                `/s/${settings.networkSecret}`);
+            toggleNewNetworkId.onTextCallback(settings.networkUUID);
+            toggleNewSecret.onTextCallback(settings.networkSecret);
+        }
+        cachedSettings = settings;
+        if (anyChanged) {
+            document.getElementById("settingsIframe").contentWindow.postMessage(JSON.stringify({
+                getSettings: realityEditor.gui.settings.generateGetSettingsJsonMessage(),
+                getMainDynamicSettings: realityEditor.gui.settings.generateDynamicSettingsJsonMessage(realityEditor.gui.settings.MenuPages.MAIN)
+            }), "*");
+        }
+    }
+
     // If we're viewing this on localhost we can connect to and read settings
     // from the local server
     if (window.location.host.split(':')[0] === localSettingsHost.split(':')[0]) {
-        setInterval(async () => {
-            let settings;
-            try {
-                let res = await fetch(`http://${localSettingsHost}/hardwareInterface/edgeAgent/settings`);
-                settings = await res.json();
-            } catch (_e) {
-                return;
-            }
-            let anyChanged = Math.random() < 0.1;
-            if (cachedSettings.isConnected !== settings.isConnected) {
-                toggleCloudUrl.onToggleCallback(settings.isConnected);
-                anyChanged = true;
-            }
-            if ((cachedSettings.serverUrl !== settings.serverUrl) ||
-                (cachedSettings.networkUUID !== settings.networkUUID) ||
-                (cachedSettings.networkSecret !== settings.networkSecret)) {
-                anyChanged = true;
-                toggleCloudUrl.onTextCallback(`https://${settings.serverUrl}/stable` +
-                                              `/n/${settings.networkUUID}` +
-                                              `/s/${settings.networkSecret}`);
-                toggleNewNetworkId.onTextCallback(settings.networkUUID);
-                toggleNewSecret.onTextCallback(settings.networkSecret);
-            }
-            cachedSettings = settings;
-            if (anyChanged) {
-                document.getElementById("settingsIframe").contentWindow.postMessage(JSON.stringify({
-                    getSettings: realityEditor.gui.settings.generateGetSettingsJsonMessage(),
-                    getMainDynamicSettings: realityEditor.gui.settings.generateDynamicSettingsJsonMessage(realityEditor.gui.settings.MenuPages.MAIN)
-                }), "*");
-            }
-        }, 1000);
+        let settings;
+        try {
+            let res = await fetch(`http://${localSettingsHost}/hardwareInterface/edgeAgent/settings`);
+            settings = await res.json();
+            processNewSettings(settings);
+        } catch (_e) {
+            return;
```

Comment:
I think this block would be better as a IIFE. Right now it will pause the whole onload flow to fetch settings and if it encounters an error it'll break the entire userinterface by returning

Reply:
Ah, you're right, I missed that when pulling it out of setInterval